### PR TITLE
Fix debugging return value inspection and continue after exception scenarios

### PR DESCRIPTION
**Pull request by agarwal-sandeep**
During redeferal change a check was removed, this broke return value
inspection and continue after exception scenario for builtins and
external functions. Basically we for debugging we wrap non script functions
in a try catch for continue after exception.

If CSP policy says eval is disabled, we should call the right entrypoints during debugging

### CODE REVIEW

**Comment by agarwal-sandeep** [comment]
@pleath @akroshg please take a look

File: lib/Runtime/Base/ScriptContext.h [hunk 1]
```diff
@@ -806,6 +806,7 @@ namespace Js
         // this is to indicate the actual close is called once only.
         bool isScriptContextActuallyClosed;
         bool isFinalized;
+        bool isEvalRestrict;
```
Comment (akroshg) [inline lib/Runtime/Base/ScriptContext.h]: >isEvalRestrict [](start = 13, length = 14)

nit: isEvalRestricted

**Comment by akroshg** [comment]
LGTM

---
In reply to: [274124377](https://github.com/Microsoft/ChakraCore/pull/2380#issuecomment-274124377) [](ancestors = 274124377)

**Comment by akroshg** [comment]
:shipit:

**Pull request merged**